platform/FilePath.cpp

```diff
@@ -18,9 +18,10 @@
 namespace mbed {
 
 FilePath::FilePath(const char* file_path) : file_name(NULL), fb(NULL) {
-    if ((file_path[0] != '/') || (file_path[1] == 0)) return;
+    // skip slashes
+    file_path += strspn(file_path, "/");
 
-    const char* file_system = &file_path[1];
+    const char* file_system = file_path;
```

Comment:
Out of curiosity, why is this a `const char*` instead of `char*`?

Reply:
Because file_path is a `const char*`. To change to `char *` would break constness. But this is describing the data, not the pointer itself. If you wanted a const _pointer_ it'd look like this `const char *const file_path`, and then you would be prevented from modifying the pointer itself. Isn't it beautiful?